pkg/pod/status.go

```diff
@@ -116,36 +115,18 @@ func MakeTaskRunStatus(tr v1alpha1.TaskRun, pod *corev1.Pod, taskSpec v1alpha1.T
 		_ = logger.Sync()
 	}()
 
+	td, err := termination.ParseMessages(pod.Status)
+	if err != nil {
+		logger.Errorf("Error parsing termination messages, start times will be inaccurate: %v", err)
+	}
+	// Overwrite container start times with termination-message-reported start times.
+	for i := range td.StartTimes {
+		if pod.Status.ContainerStatuses[i].State.Terminated != nil {
+			pod.Status.ContainerStatuses[i].State.Terminated.StartedAt = td.StartTimes[i]
```

Comment:
It's a tad confusing that we're modifying the pod here but those changes won't be written back to kubernetes (and therefore these changes won't show up for someone kubectl describing the pod).  Maybe consider putting the container statuses into a named variable or adding a comment here?